chevah/brink
# [#4071] Fix Unicode path passing on Windows/Mac OSX.

Scope
=====

The following error was displayed when running paver on Windows:

```
Exception:
Traceback (most recent call last):
  File "d:\_Work\chevah\server\build-windows-x86\lib\lib\site-packages\pip\basecommand.py", line 215, in main
    status = self.run(options, args)
  File "d:\_Work\chevah\server\build-windows-x86\lib\lib\site-packages\pip\commands\install.py", line 342, in run
    prefix=options.prefix_path,
  File "d:\_Work\chevah\server\build-windows-x86\lib\lib\site-packages\pip\req\req_set.py", line 784, in install
    **kwargs
  File "d:\_Work\chevah\server\build-windows-x86\lib\lib\site-packages\pip\req\req_install.py", line 851, in install
    self.move_wheel_files(self.source_dir, root=root, prefix=prefix)
  File "d:\_Work\chevah\server\build-windows-x86\lib\lib\site-packages\pip\req\req_install.py", line 1064, in move_wheel_files
    isolated=self.isolated,
  File "d:\_Work\chevah\server\build-windows-x86\lib\lib\site-packages\pip\wheel.py", line 272, in move_wheel_files
    compileall.compile_dir(source, force=True, quiet=True)
  File "d:\_Work\chevah\server\build-windows-x86\lib\lib\compileall.py", line 57, in compile_dir
    quiet):
  File "d:\_Work\chevah\server\build-windows-x86\lib\lib\compileall.py", line 57, in compile_dir
    quiet):
  File "d:\_Work\chevah\server\build-windows-x86\lib\lib\compileall.py", line 57, in compile_dir
    quiet):
  File "d:\_Work\chevah\server\build-windows-x86\lib\lib\compileall.py", line 57, in compile_dir
    quiet):
  File "d:\_Work\chevah\server\build-windows-x86\lib\lib\compileall.py", line 57, in compile_dir
    quiet):
  File "d:\_Work\chevah\server\build-windows-x86\lib\lib\compileall.py", line 50, in compile_dir
    if not compile_file(fullname, ddir, force, rx, quiet):
  File "d:\_Work\chevah\server\build-windows-x86\lib\lib\compileall.py", line 102, in compile_file
    print 'Compiling', fullname, '...'
TypeError: 'unicode' does not have the buffer interface
```

Changes
=======

For `Windows` and `Mac OSX` we are passing UTF-8 encoded values instead of Unicode.

How to try and test the changes
===============================

reviewers: @adiroiban @lgheorghiu @hcs0 

Please check that the changes make sense and the problem is not longer reproducible.

I have tested this on my machine and it works.
